### PR TITLE
Update trustyAI test_notebooks 

### PR DESCRIPTION
**Pull request by atheo89**
This is a follow up PR for this:  https://github.com/opendatahub-io/notebooks/pull/510
 
Update trustyAI test_notebook to include the tests from minimal and datascience due to python package inconsistencies

## How Has This Been Tested?
Tested the ubi9 test_notebok.ipynb flavor in ODH using TrustyAI 2024.1 workbench

You should see the following results:
```
test_create_default_session (__main__.TestBoto3) ... ok
test_version (__main__.TestBoto3) ... ok
test_jupyter_version (__main__.TestDependenciesVersions) ... ok
test_nbdime_version (__main__.TestDependenciesVersions) ... ok
test_nbgitpuller_version (__main__.TestDependenciesVersions) ... ok
test_buffer_pool (__main__.TestKafkaPython) ... ok
test_session_timeout_larger_than_request_timeout_raises (__main__.TestKafkaPython) ... ok
test_version (__main__.TestKafkaPython) ... ok
test_matplotlib_figure_creation (__main__.TestMatplotlib) ... ok
test_version (__main__.TestMatplotlib) ... ok
test_array_creation (__main__.TestNumpy) ... ok
test_array_opeartions (__main__.TestNumpy) ... ok
test_array_statistical_functions (__main__.TestNumpy) ... ok
test_version (__main__.TestNumpy) ... ok
test_dataframe_creation (__main__.TestPandas) ... ok
test_dataframe_shape (__main__.TestPandas) ... ok
test_drop (__main__.TestPandas) ... ok
test_equal_dataframes (__main__.TestPandas) ... ok
test_index_out_of_bounds (__main__.TestPandas) ... ok
test_sampling (__main__.TestPandas) ... ok
test_unequal_dataframes (__main__.TestPandas) ... ok
test_version (__main__.TestPandas) ... ok
test_version (__main__.TestPythonVersion) ... ok
test_sklearn_dataset (__main__.TestSKLearn) ... ok
test_sklearn_train_test_split (__main__.TestSKLearn) ... ok
test_version (__main__.TestSKLearn) ... ok
test_scipy_integrate (__main__.TestScipy) ... ok
test_scipy_special (__main__.TestScipy) ... ok
test_version (__main__.TestScipy) ... ok
test_datafairness (__main__.TestTrustyaiNotebook) ... ok
test_fairnessmetrics (__main__.TestTrustyaiNotebook) ... ok
test_trustyai_version (__main__.TestTrustyaiNotebook) ... ok

```

Or should wait the CI to build the two trustyAI workbenches and then run 
`podman run -it --rm -p 8888:8888 ${TrustyAI_CI_notebook}`
and run inside the test_notebook.ipynb (ubi8 & ubi9)

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb [hunk 1]
```diff
@@ -8,26 +8,241 @@
...
+    "class TestTrustyaiNotebook(unittest.TestCase):\n",
+    "\n",
+    "    def test_trustyai_version(self): \n",
+    "        expected_major_minor = '0.6' # Set the expected version (x.y) \n",
+    "        actual_major_minor = get_major_minor(trustyai.__version__)\n",
+    "        self.assertEqual(actual_major_minor, expected_major_minor, \"incorrect version\")\n",
+    "\n",
     "    def test_fairnessmetrics(self):\n",
-    "        url_unbiased = \"https://raw.githubusercontent.com/opendatahub-io/notebooks/main/jupyter/trustyai/ubi8-python-3.8/test/income-unbiased.csv\"\n",
+    "        url_unbiased = \"https://raw.githubusercontent.com/opendatahub-io/notebooks/main/jupyter/trustyai/ubi9-python-3.9/test/income-unbiased.csv\"\n",
```
Comment (jstourac) [inline jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb:245]: Is this change intentional?

File: jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb [hunk 2]
```diff
@@ -44,7 +259,7 @@
     "        print(\"On the test_fairness_metrics test case the statistical_parity_difference score for this dataset is between the threshold [-0.1,0.1], which classifies the model as reasonably fair.\")\n",
     "        \n",
     "    def test_datafairness(self):\n",
-    "        url_biased = \"https://raw.githubusercontent.com/opendatahub-io/notebooks/main/jupyter/trustyai/ubi8-python-3.8/test/income-biased.csv\"\n",
+    "        url_biased = \"https://raw.githubusercontent.com/opendatahub-io/notebooks/main/jupyter/trustyai/ubi9-python-3.9/test/income-biased.csv\"\n",
```
Comment (jstourac) [inline jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb:262]: same as above?

File: jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb (hunk 1 above)
Comment (atheo89) [reply jupyter/trustyai/ubi8-python-3.8/test/test_notebook.ipynb]: good catch! Just fixed

**Review by jstourac — APPROVED** [review]
